app/controllers/api/connections_controller.rb

```diff
@@ -10,15 +10,24 @@ def request_match
 
             if SkillClass.find_by(id: params[:id])
                 #check if class exists
-                request.class_id = params[:id]
+                request.skill_class_id = params[:id]
 
-                if MatchRequest.find_by(student_id: @current_user.id, class_id: params[:id])
-                    #check if request already exists
+                most_recent_request = MatchRequest.where(student_id: @current_user.id, skill_class_id: params[:id]).order(:created_at).reverse_order.first
```

Comment:
You can define the direction of the sort in the `order` call. Just change `order(:created_at)` to `order(created_at: :desc)` and you can remove that `.reverse_order` from there